src/Ztm.WebApi/Watchers/TransactionConfirmation/EntityWatchRepository.cs

```diff
@@ -0,0 +1,113 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using Ztm.Data.Entity.Contexts;
+using Ztm.Data.Entity.Contexts.Main;
+using Ztm.Zcoin.Watching;
+
+namespace Ztm.WebApi.Watchers.TransactionConfirmation
+{
+    public sealed class EntityWatchRepository : IWatchRepository
+    {
+        readonly IMainDatabaseFactory db;
+
+        public EntityWatchRepository(IMainDatabaseFactory db)
+        {
+            if (db == null)
+            {
+                throw new ArgumentNullException(nameof(db));
+            }
+
+            this.db = db;
+        }
+
+        public async Task AddAsync(TransactionWatch<Rule> watch, CancellationToken cancellationToken)
+        {
+            if (watch == null)
+            {
+                throw new ArgumentNullException(nameof(watch));
+            }
+
+            if (watch.Context == null)
+            {
+                throw new ArgumentNullException(nameof(watch.Context));
```

Comment:
We should not do this due to argument name should not refer to the members of argument. Instead, use `ArgumentException` to tell that `Context` is `null`.